### PR TITLE
[Method References] LPS-161243 Replace method setSamlProviderConfigurationHelper with variable samlProviderConfigurationHelper in class extends BaseSamlStrutsAction

### PR DESCRIPTION
**Pull request by W2jx**
Hi @tinatian @dantewang , I have replaced method setSamlProviderConfigurationHelper with variable samlProviderConfigurationHelper in this pr.

### CODE REVIEW

**Comment by W2jx** [comment]
Hi @catarineaguiar , Could you please help me check the test results when they finish ? Thanks!

**Comment by W2jx** [comment]
no conflicts(6:16 PM)

File: modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/BaseSamlStrutsAction.java [hunk 1]
```diff
@@ -92,21 +92,14 @@ public String execute(
...
 
 	public boolean isEnabled() {
-		return samlProviderConfigurationHelper.isEnabled();
-	}
-
-	public void setSamlProviderConfigurationHelper(
-		SamlProviderConfigurationHelper samlProviderConfigurationHelper) {
-
-		this.samlProviderConfigurationHelper = samlProviderConfigurationHelper;
+		return getSamlProviderConfigurationHelper().isEnabled();
```
Comment (dantewang) [inline modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/BaseSamlStrutsAction.java:95]: Hmm this passes SF..

File: modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/WebSsoAction.java [hunk 2]
```diff
@@ -63,6 +61,9 @@ protected String doExecute(
 		return null;
 	}
 
+	@Reference(unbind = "-")
```
Comment (dantewang) [inline modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/WebSsoAction.java:64]: @W2jx Usually we don't add `unbind = "-"` to field reference, please clean it up.

**Review by dantewang — CHANGES_REQUESTED** [review]
@W2jx Please see my comment and make the change. Others look good to me.

**Comment by catarineaguiar** [comment]
@W2jx no regressions found.

**Comment by W2jx** [comment]
@catarineaguiar Thank you

**Comment by W2jx** [comment]
Push the code just to fix the problem in the comment.

File: modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/WebSsoAction.java (hunk 2 above)
Comment (W2jx) [reply modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/WebSsoAction.java]: Thank you, I will fix it.

**Comment by W2jx** [comment]
no conflicts(6:50 AM)

File: modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/BaseSamlStrutsAction.java [hunk 3]
```diff
@@ -88,23 +88,18 @@ public String execute(
 		return null;
 	}
 
-	public boolean isEnabled() {
-		return samlProviderConfigurationHelper.isEnabled();
-	}
-
-	public void setSamlProviderConfigurationHelper(
-		SamlProviderConfigurationHelper samlProviderConfigurationHelper) {
+	public abstract SamlProviderConfigurationHelper
```
Comment (tinatian) [inline modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/BaseSamlStrutsAction.java:91]: @W2jx Why this is a public method? This is an internal class, no one is going to use this public method.

File: modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/BaseSamlStrutsAction.java (hunk 1 above)
Comment (tinatian) [reply modules/dxp/apps/saml/saml-web/src/main/java/com/liferay/saml/web/internal/struts/BaseSamlStrutsAction.java:95]: @W2jx Do not do this even if it passed SF.

**Comment by tinatian** [comment]
@W2jx @dantewang This samlProviderConfigurationHelper is only used by isEnabled(), please inline the logic inside isEnabled to avoid adding getSamlProviderConfigurationHelper().

Tina

**Comment by W2jx** [comment]
@tinatian Thank you, and I will fix it.

**Pull request closed without merging**